arangod/Cluster/UpdateReplicatedLogAction.cpp

```diff
@@ -0,0 +1,100 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2021-2021 ArangoDB GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Lars Maier
+////////////////////////////////////////////////////////////////////////////////
+
+#include <optional>
+
+#include "ApplicationFeatures/ApplicationServer.h"
+#include "Basics/Exceptions.h"
+#include "Basics/StringUtils.h"
+#include "Cluster/MaintenanceFeature.h"
+#include "Network/NetworkFeature.h"
+#include "Replication2/ReplicatedLog/AgencyLogSpecification.h"
+#include "Replication2/ReplicatedLog/Algorithms.h"
+#include "Replication2/ReplicatedLog/NetworkAttachedFollower.h"
+#include "Replication2/ReplicatedLog/ReplicatedLog.h"
+#include "RestServer/DatabaseFeature.h"
+#include "UpdateReplicatedLogAction.h"
+#include "Utils/DatabaseGuard.h"
+
+using namespace arangodb::basics;
+using namespace arangodb::replication2;
+
+
+
+bool arangodb::maintenance::UpdateReplicatedLogAction::first() {
+
+  struct LogActionContextMaintenance : algorithms::LogActionContext {
+    LogActionContextMaintenance(TRI_vocbase_t& vocbase, network::ConnectionPool* pool)
+        : vocbase(vocbase), pool(pool) {}
+
+    auto dropReplicatedLog(LogId id) -> arangodb::Result override {
+      return vocbase.dropReplicatedLog(id);
+    }
+    auto ensureReplicatedLog(LogId id)
+        -> std::shared_ptr<replicated_log::ReplicatedLog> override {
+      return vocbase.ensureReplicatedLog(id, std::nullopt);
+    }
+    auto buildAbstractFollowerImpl(LogId id, ParticipantId participantId)
+    -> std::shared_ptr<replicated_log::AbstractFollower> override {
+      return std::make_shared<replicated_log::NetworkAttachedFollower>(pool, std::move(participantId), vocbase.name(), id);
+    }
+
+    TRI_vocbase_t& vocbase;
+    network::ConnectionPool* pool;
+  };
+
+  auto spec = std::invoke([&]() -> std::optional<agency::LogPlanSpecification> {
+    auto buffer = StringUtils::decodeBase64(_description.get(REPLICATED_LOG_SPEC));
```

Comment:
Wondering why we are base64-encoding/-decoding the log specification. I know that the VelocyPack value would otherwise be binary and contain whatever non-printable characters. But is that actually a problem?

Reply:
Some people say that [std::string is not a Container for Raw Data](https://arne-mertz.de/2018/11/string-not-for-raw-data/)

Reply:
https://www.youtube.com/watch?v=SDJImePyftY

Reply:
LOL